src/strategies/property.sml

```diff
@@ -6,45 +6,91 @@ import "strategies.property_importance";
 
 signature PROPERTY =
 sig
+    exception ParseError;
+
+    type pKind;
+    datatype pValue = Label of string | Number of int | Boolean of bool;
     type property;
 
-    exception ParseError;
+    val stringOfpKind : pKind -> string;
+    val pKindOfString : string -> pKind;
+
+    val kindOf : property -> pKind;
+    val valueOf : property -> pValue;
+    val typeOf : property -> Type.T;
+    val attributesOf : property -> string list;
 
     val compare : property * property -> order;
     val match : property * property -> bool;
-    val nameOf : property -> string;
+
     val toString : property -> string;
+    val fromKindValuePair : pKind * pValue -> property;
```

Comment:
Should `fromKindValuePair` have a dual `toKindValuePair`?

Reply:
done